packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -533,7 +533,13 @@ List<AndroidDevice> getAdbDevices() {
   final String adbPath = getAdbPath(androidSdk);
   if (adbPath == null)
     return <AndroidDevice>[];
-  final String text = runSync(<String>[adbPath, 'devices', '-l']);
+  String text;
+  try {
+    text = runSync(<String>[adbPath, 'devices', '-l']);
+  } on ArgumentError catch (exception) {
+    throwToolExit('Unable to run "adb", check your Android SDK installation and '
```

Comment:
Can we trust that `exception.message` will contain `adbPath`, or should we explicitly include that path in the error message here?

Reply:
It does include it, but I'll test for that in the test to make sure it doesn't change.

The error looks like this in the wild:
```
Unable to run "adb", check your Android SDK installation and ANDROID_HOME environment variable: Cannot find executable for /usr/local/google/home/gspencer/Android/Sdk/platform-tools/adb.
```